builder/dockerfile/copy.go

```diff
@@ -315,6 +317,10 @@ func (o *copier) copyWithWildcards(origPath string) ([]copyInfo, error) {
 func copyInfoForFile(source builder.Source, path string) (copyInfo, error) {
 	fi, err := remotecontext.StatAt(source, path)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// return the relative path in the error, which is more user-friendly than the full path to the tmp-dir
+			return copyInfo{}, errors.WithStack(&os.PathError{Op: "stat", Path: path, Err: os.ErrNotExist})
```

Comment:
nit: Is non-os pkg allowed to return `os.PathError`?

Reply:
Not sure; I know we do it in other places already, e.g. https://github.com/moby/moby/blob/3aac5f0bbb5ccaeb9081ef710aae2459cbc87fa7/pkg/idtools/idtools_unix.go#L37